### PR TITLE
use product overrides in web

### PR DESCRIPTION
**Pull request by sandy081**

### CODE REVIEW

**Review by aeschli — DISMISSED** [review]

**Comment by aeschli** [comment]
@sandy081 I wonder if I need to adopt @vscode-test/web as well
https://github.com/microsoft/vscode-test-web/blob/6d4145d916ca746f9af5fe1bc0f8c15da965d55f/views/workbench.html

**Comment by sandy081** [comment]
@aeschli Adopted in vscode-test-web module.

File: src/vs/platform/product/common/product.ts [hunk 1]
```diff
@@ -55,21 +55,6 @@ else {
 	// Built time configuration (do NOT modify)
 	product = { /*BUILD->INSERT_PRODUCT_CONFIGURATION*/ } as IProductConfiguration;
 
-	// Running out of sources
```
Comment (bpasero) [inline src/vs/platform/product/common/product.ts:58]: I am not sure we can just remove this block, if any code depends on `product.ts` directly (and not product service), the behaviour would be different now?

File: src/vs/code/browser/workbench/workbench-dev.html [hunk 2]
```diff
@@ -25,6 +25,9 @@
 		<!-- Builtin Extensions (running out of sources) -->
 		<meta id="vscode-workbench-builtin-extensions" data-settings="{{WORKBENCH_BUILTIN_EXTENSIONS}}">
 
+		<!-- Product Overrides (running out of sources) -->
```
Comment (bpasero) [inline src/vs/code/browser/workbench/workbench-dev.html:28]: I am curious why we do this actually only for running out of sources (also on desktop)?

File: src/vs/workbench/browser/web.main.ts [hunk 3]
```diff
@@ -542,3 +543,33 @@ export class BrowserMain extends Disposable {
 		return UNKNOWN_EMPTY_WINDOW_WORKSPACE;
 	}
 }
+
+function getProductConfiguration(): IProductConfiguration {
```
Comment (bpasero) [inline src/vs/workbench/browser/web.main.ts:547]: This is not a good location for the method. You depend on a `vscode-workbench-product-overrides` element which will only ever be present when our server and our web HTML runs, but for example not on github.dev. If at all this code should go into `src/vs/code/browser/workbench/workbench.ts` which is the specific entry point that gets triggered from our web UI.

**Review by bpasero — CHANGES_REQUESTED** [review]
I still fail to understand why we do this and why its only for out of sources.

**Comment by sandy081** [comment]
@bpasero Thanks for the feedback. Sorry for not giving the context. 

This is to help developers easily override product configuration without touching the `product.json` in sources. Existing methodology is a quite bit of hassle especially making sure the changes done to product.json locally for debugging/testing purposes are not checked in. Hence this new methodology - having a product.overrides.json in sources that is git ignored and is loaded only in dev - will help a ton for developers. 

Regarding your feedback on changes, I liked your idea of using existing `productConfiguration` embedder API. Updated the PR accordingly. Please check

**Review by bpasero — APPROVED** [review]
Elegant :thumbsup:

**Pull request merged**